packages/core-ethereum/src/indexer/index.mock.ts

```diff
@@ -0,0 +1,239 @@
+      return pastEvents
+    }
+  }
+
+  const hoprChannels = new FakeChannels() as unknown as HoprChannels
+
+  return {
+    hoprChannels,
+    pubkeys,
+    newEvent(event: Event<any>) {
+      pastEvents.push(event)
+      hoprChannels.emit('*', event)
+    }
+  }
+}
+const createHoprTokenMock = () => {
+  class FakeToken extends EventEmitter {
+    async transfer() {
+      let newEvent = {
+        event: 'Transfer',
+        transactionHash: '',
+        blockNumber: 8,
+        transactionIndex: 0,
+        logIndex: 0,
+        args: {
+          source: PARTY_A.toAddress().toHex(),
+          destination: PARTY_B.toAddress().toHex(),
+          balance: BigNumber.from('1')
+        } as any
+      } as TokenEvent<'Transfer'>
+      this.emit('*', newEvent)
+    }
+  }
+
+  const hoprToken = new FakeToken() as unknown as HoprToken
+
+  return {
+    hoprToken,
+    newEvent(event: Event<any>) {
+      hoprToken.emit('*', event)
+    }
+  }
+}
+
+const createChainMock = (
+  provider: Providers.WebSocketProvider,
+  hoprChannels: HoprChannels,
+  hoprToken: HoprToken,
+  account?: Wallet
+): ChainWrapper => {
+  return {
+    getLatestBlockNumber: () => provider.getBlockNumber(),
+    subscribeBlock: (cb) => provider.on('block', cb),
+    subscribeError: (cb) => {
+      provider.on('error', cb)
+      hoprChannels.on('error', cb)
+    },
+    subscribeChannelEvents: (cb) => hoprChannels.on('*', cb),
+    start: () => {},
+    waitUntilReady: () => {
+      chainLogger('Await on chain readyness')
+      return Promise.resolve()
+    },
+    getGenesisBlock: () => {
+      chainLogger('Genesis log requested')
+      return 0
+    },
+    waitForPublicNodes: () => {
+      chainLogger('On-chain request for existing public nodes.')
+      return Promise.resolve([])
+    },
+    announce: () => {
+      chainLogger('On-chain announce request sent')
+    },
+    on: (event: string) => {
+      chainLogger(`On-chain signal for event "${event}"`)
+    },
+    indexer: {
+      on: (event: string) => chainLogger(`Indexer on handler top of chain called with event "${event}"`),
+      off: (event: string) => chainLogger(`Indexer off handler top of chain called with event "${event}`)
+    },
+    subscribeTokenEvents: (cb) => hoprToken.on('*', cb),
+    getNativeTokenTransactionInBlock: (_blockNumber: number, _isOutgoing: boolean = true) => [],
+    updateConfirmedTransaction: (_hash: string) => {},
+    unsubscribe: () => {
+      provider.removeAllListeners()
+      hoprChannels.removeAllListeners()
+    },
+    getNativeBalance: () => new NativeBalance(SUGGESTED_NATIVE_BALANCE),
+    getChannels: () => hoprChannels,
+    getWallet: () => account ?? fixtures.ACCOUNT_A,
+    getAccount: () => {
+      chainLogger('getAccount method was called')
+      return Promise.resolve(
+        new AccountEntry(
+          fixtures.PARTY_A.toAddress(),
+          new Multiaddr(`/ip4/127.0.0.1/tcp/124/p2p/${fixtures.PARTY_A.toB58String()}`),
```

Comment:
this adds implications. Why don't pass the id in the constructor?

Reply:
Same here.